<<<<<<< HEAD
syntax = "proto3";
package Protobuf;
option csharp_namespace = "Communication.Proto";
enum ObjType{
    player = 1;
    wall = 2;
    prop = 3;
    bullet = 4;


}

enum PropType{
    prop1 = 1;
    prop2 = 2;
    prop3 = 3;

}

enum BulletType{
    bullet1 = 1;
    bullet2 = 2;
    bullet3 = 3;
 
}

enum BuffType{
    haveShield = 1;
    haveFuhuojia = 2;
    speedUp = 3;
    speedDown = 4;
    damageUp = 5;
    damageDown = 6;
    isStunned = 7;

}

enum JobType {
    job1 = 1;
    job2 = 2;
    job3 = 3;

}

message GameObject{
    double x = 1;
    double y = 2;
    int hp = 3;
    int moveSpeed = 4;
    int bulletNum = 5;
    ObjType objType = 6;//
    PropType propType = 7;//
    BulletType bulletType = 8;//
    JobType jobType = 9;
    int teamId = 10;
    int playerId = 11;//区分队内id
    int damage = 12;
    // bool haveShield;
    // bool haveFuhuojia;
    // bool speedUp;
    // bool speedDown;
    // bool damageUp; 	//是否有攻击力加成
    // bool damageDown; //是否攻击力减慢
    // bool isStunned; //是否处于眩晕状态
    double facingAngle = 13; //面对的方向,rad
    repeated BuffType buffs = 14;
}

message MapColor{
    int x = 1;
    int y = 2;
    int teamId = 3;
}

message Message2Client{
    map <int64, GameObject> gameObjectList = 1; 
    repeated MapColor mapColors = 2;

=======
syntax = "proto3";
package Protobuf;
option csharp_namespace = "Communication.Proto";

enum ObjType{
    player = 0;
    wall = 1;
    prop = 2;
    bullet = 3;
}
enum PropType{
    prop1 = 0;
    prop2 = 1;
    prop3 = 2;
}
enum BulletType{
    bullet1 = 0;
    bullet2 = 1;
    bullet3 = 2;
}
enum BuffType{
    haveShield = 0;
    haveFuhuojia = 1;
    speedUp = 2;
    speedDown = 3;
    damageUp = 4;
    damageDown = 5;
    isStunned = 6;
}
enum JobType {
    job1 = 0;
    job2 = 1;
    job3 = 2;
}
message GameObject{
    double x = 1;
    double y = 2;
    int32 hp = 3;
    int32 moveSpeed = 4;
    int32 bulletNum = 5;
    ObjType ObjType = 6;//
    PropType PropType = 7;//
    BulletType BulletType = 8;//
    JobType JobType = 9;
    int32 teamId = 10;
    int32 playerId = 11;//区分队内id
    int32 damage = 12;
    double facingAngle = 13; //面对的方向,rad
    repeated BuffType buffs = 14;
}

message MapColor{
    int32 x = 1;
    int32 y = 2;
    int32 teamId = 3;
}

message Message2Client{
    map <int64, GameObject> gameObjectList = 1; 
    repeated MapColor MapColors = 2;
>>>>>>> c1014fd5771cbdda923ce230fe65180c4d66f406
}